tasks/gulp/compile-assets.mjs

```diff
@@ -129,30 +129,61 @@ export async function compileJavaScripts () {
       ? componentNameToJavaScriptModuleName(name)
       : 'GOVUKFrontend'
 
-    return gulp.src(slash(join(configPaths.src, file)))
-      .pipe(rollup({
-        // Used to set the `window` global and UMD/AMD export name
-        // Component JavaScript is given a unique name to aid individual imports, e.g GOVUKFrontend.Accordion
-        name: moduleName,
-        // Legacy mode is required for IE8 support
-        legacy: true,
-        // UMD allows the published bundle to work in CommonJS and in the browser.
-        format: 'umd'
-      }))
-      .pipe(gulpif(isDist, uglify({
-        ie8: true
-      })))
-      .pipe(gulpif(isDist,
-        rename({
-          basename: 'govuk-frontend',
-          extname: '.min.js'
-        })
-      ))
-      .pipe(rename({
-        extname: '.js'
-      }))
+    return compileJavaScript(gulp.src(slash(join(configPaths.src, file))), moduleName)
       .pipe(gulp.dest(slash(join(destPath, modulePath))))
   }))
 }
 
 compileJavaScripts.displayName = 'compile:js'
+
+/**
+ * Compile JavaScript ESM to CommonJS helper
+ *
+ * @param {import('stream').Stream} stream - Input file stream
+ * @param {string} moduleName - Rollup module name
+ * @returns {import('stream').Stream} Output file stream
+ */
+function compileJavaScript (stream, moduleName) {
+  return stream
+    .pipe(plumber(function (cause) {
+      const error = new PluginError('compile:js', cause)
+      console.error(error.toString())
+
+      // Gulp continue (watch)
+      if (isDev) {
+        return this.emit('end')
+      }
+
+      // Gulp exit with error
+      return stream.emit('error', error)
```

Comment:
**question** This looks slightly different from the one used in #3034:
- we're building the `PluginError` with `cause` instead of `cause.messageFormatted`
- we're not doing an `error.showProperties = false`
- we're `console.error`ing the error in all cases, rather than just in dev

Are these differences intentional or just due to things having been written at different times?